File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/AddressTypeRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface AddressTypeRow {
    AddressTypeId?: number;
    AddressTypeName?: string;
    IsActive?: number;
    TenantId?: number;
}

export abstract class AddressTypeRow {
    static readonly idProperty = 'AddressTypeId';
    static readonly nameProperty = 'AddressTypeName';
    static readonly localTextPrefix = 'Company.AddressType';

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<AddressTypeRow>();
}

File: 2025/Company/Company.Web/Modules/Company/AddressType/AddressTypeRow.cs
using Company.Administration.Entities;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[person].[AddressType]")]
[DisplayName("Address Type"), InstanceName("Address Type")]
[ReadPermission(PermissionKeys.AddressType.View)]
[ModifyPermission(PermissionKeys.AddressType.Modify)]
[DeletePermission(PermissionKeys.AddressType.Delete)]
[ServiceLookupPermission("Company:General")]
[LookupScript("Company.AddressType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class AddressTypeRow : LoggingRow<AddressTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Address Type Id"), Identity, IdProperty]
    public int? AddressTypeId { get => fields.AddressTypeId[this]; set => fields.AddressTypeId[this] = value; }

    [DisplayName("Address Type Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string AddressTypeName { get => fields.AddressTypeName[this]; set => fields.AddressTypeName[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field AddressTypeId;
        public StringField AddressTypeName;

        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}